041-opa-integration/proposal.md

```diff
@@ -0,0 +1,147 @@
+# Summary
+
+This proposal outlines the beginnings of support for policy enforcement. OPA 
+(Open Policy Agent) will be used as the first policy manager to integrate with.
+
+
+# Motivation
+
+Generally speaking, anything that exposes an HTTP API (whether an individual 
+micro-service or an application as a whole) needs to control who can run those 
+APIs and when. 
+
+Concourse, as a CI platform, also needs to control who can do something and when,
+which can be done by integrating with a policy manager. Before any action, send
+a policy check request, with data that describes the action, to the policy manager,
+and only continue if the policy manager replies a "pass".
+
+Possible policies to apply could be:
+
+* Steps, e.g. code security scan, by certain resource types must be done.
+* Some team's tasks must be run with certain tag.
+* Some docker image registries are not allowed.
+* Some docker images are not not allowed.
+* and so on ...
+
+
+# Proposal
+
+## Support multiple policy managers
+
+Like how multiple credential managers are supported, Concourse should allow other
+policy managers than OPA.
```

Comment:
This sounds reasonable, but I think the RFC should show how this would work or leave it as an open question.

Reply:
Added a description.